sql/catalyst/src/main/java/org/apache/spark/sql/util/CaseInsensitiveStringMap.java

```diff
@@ -40,9 +40,12 @@ public static CaseInsensitiveStringMap empty() {
     return new CaseInsensitiveStringMap(new HashMap<>(0));
   }
 
+  private final Map<String, String> original;
+
   private final Map<String, String> delegate;
 
   public CaseInsensitiveStringMap(Map<String, String> originalMap) {
+    this.original = new HashMap<>(originalMap);
```

Comment:
this should be `new HashMap<>(originalMap.size);`, otherwise we add data to it twice.